Source/Core/Matrix3.js

```diff
@@ -1023,6 +1023,27 @@ define([
         return result;
     };
 
+    var UNIT = new Cartesian3(1, 1, 1);
+    var scaleVector = new Cartesian3();
+
+    /**
+     * Factors rotation matrix from arbitrary Matrix3 by dividing components by scale
+     *
+     * @param {Matrix3} rotation-scale matrix
+     * @param {Matrix3} result rotation matrix with unit scale
+     * @returns {Matrix3} The modified result parameter.
+     */
+    Matrix3.getRotation = function(matrix, result) {
+        Matrix3.multiplyByScale(
```

Comment:
Check that the arguments are defined:

```
        //>>includeStart('debug', pragmas.debug);
        Check.typeOf.object('matrix', matrix);
        Check.typeOf.object('result', result);
        //>>includeEnd('debug');
```

Reply:
Also, this is better written as `result = Matrix3.multiplyByScale(.....`.  It makes the code more clear to someone not as familiar with Javascript who might not realize the result argument is being mutated